File: src/main/java/br/com/alura/forum/actuator/endpoints/UnansweredTopicsActuatorEndpoint.java
package br.com.alura.forum.actuator.endpoints;

import java.util.List;

import br.com.alura.forum.repository.TopicRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.actuate.endpoint.annotation.Endpoint;
import org.springframework.boot.actuate.endpoint.annotation.ReadOperation;
import org.springframework.stereotype.Component;

import br.com.alura.forum.model.OpenTopicByCategory;

@Endpoint(id = "open-topics-by-category")
@Component
public class UnansweredTopicsActuatorEndpoint {
	@Autowired
	private TopicRepository topicRepository;

	@ReadOperation
	public List<OpenTopicByCategory> execute() {
		return this.topicRepository.findOpenTopicsByCategory();
	}
}

File: src/main/kotlin/br/com/alura/forum/actuator/endpoints/UnansweredTopicsActuatorEndpoint.kt
package br.com.alura.forum.actuator.endpoints

import br.com.alura.forum.repository.TopicRepository
import org.springframework.boot.actuate.endpoint.annotation.Endpoint
import org.springframework.boot.actuate.endpoint.annotation.ReadOperation
import org.springframework.stereotype.Component

@Endpoint(id = "open-topics-by-category")
@Component
class UnansweredTopicsActuatorEndpoint(private val topicRepository: TopicRepository) {
    @ReadOperation
    fun execute() = topicRepository.findOpenTopicsByCategory()
}